### DISCUSSION TITLE
Is there debounce implemented on unicorn:keyup?

### DISCUSSION THREAD

**Discussion by zurtri** [django-commons/django-unicorn#234] [Q&A] [answered]
I was writing some code yesterday where I was calling a function on keyup.

I tried adding `unicorn:keyup-debounce-1000` but it didnt seem like debounce is implemented.

Or was I just doing it wrong?

**Comment by adamghill** [comment]
Hmm, nope I only implemented the `debounce` modifier for models, but not for actions. Is that something that would be useful for you? If so, I can make a ticket and see how easy it is to add it in.

**Reply by zurtri** [reply]
I would use the debounce modifier on Actions. 

I use it a bit when I do Livewire at work.

I hope it's not too much trouble to add.

Thanks!

**Comment by adamghill** [comment, marked as answer]
I have a PR started to add this in https://github.com/adamghill/django-unicorn/pull/246. Examples: `unicorn:click.debounce-1000` or `unicorn:click.keyup.enter.debounce-1000`. I'll let you know when this gets released!

**Reply by adamghill** [reply]
Sorry for the wait. This finally got released as in version 0.32.0!

**Reply by nerdoc** [reply]
Hi Adam, using 0.35.3, `unicorn:keyup.enter.debounce-1000`works, but `unicorn:keyup.debounce-1000` does not, it fires instantly. Should I open a bug report for that? Or do you have a quick answer? BTW: a `unicorn:click.keyup.enter.debounce-1000`does nothing neither

**Reply by adamghill** [reply]
It would be great if you opened an issue for this. I think I know where the problem might be, but I'll need to look into it more to be sure. Thanks!

**Reply by adamghill** [reply]
https://github.com/adamghill/django-unicorn/issues/279 is the issue for this functionality.
